File: frontend/talentlink_frontend/lib/services/auth_service.dart
import 'dart:convert';
import 'package:http/http.dart' as http;

class AuthService {
  // Correct endpoint for Kubernetes Ingress
  static const String _baseUrl = "http://talentlink.local/api/auth";

  Future<Map<String, dynamic>> login(String username, String password) async {
    final uri = Uri.parse("$_baseUrl/login");
    final response = await http.post(
      uri,
      headers: {"Content-Type": "application/json"},
      body: jsonEncode({"username": username, "password": password}),
    );

    if (response.statusCode == 200) {
      return jsonDecode(response.body);
    } else {
      final decoded = jsonDecode(response.body);
      throw Exception(decoded['error'] ?? 'Login failed');
    }
  }

  Future<Map<String, dynamic>> register({
    required String username,
    required String email,
    required String password,
    required String role,
  }) async {
    final uri = Uri.parse("$_baseUrl/register");
    final response = await http.post(
      uri,
      headers: {"Content-Type": "application/json"},
      body: jsonEncode({
        "username": username,
        "email": email,
        "password": password,
        "role": role,
      }),
    );

    if (response.statusCode == 201) {
      return jsonDecode(response.body);
    } else {
      final decoded = jsonDecode(response.body);
      throw Exception(decoded['error'] ?? 'Registration failed');
    }
  }

  Future<bool> checkHealth() async {
    try {
      final res = await http.get(Uri.parse("$_baseUrl/health"));
      return res.statusCode == 200;
    } catch (_) {
      return false;
    }
  }
}

File: frontend/talentlink-react/src/services/authService.ts
import axios from 'axios';

const BASE_URL = 'http://talentlink.local/api/auth';

export interface LoginResponse {
  access_token: string;
  refresh_token: string;
  token_type: string;
  expires_in: number;
}

export interface RegisterRequest {
  username: string;
  email: string;
  password: string;
  role: string;
}

export class AuthService {
  static async login(username: string, password: string): Promise<LoginResponse> {
    try {
      const response = await axios.post<LoginResponse>(`${BASE_URL}/login`, {
        username,
        password,
      });
      return response.data;
    } catch (error: any) {
      const message = 
        error.response?.data?.error || 
        error.message || 
        'Login failed - please check your credentials';
      console.error('AuthService login error:', error);
      throw new Error(message);
    }
  }

  static async register(data: RegisterRequest): Promise<void> {
    try {
      await axios.post(`${BASE_URL}/register`, data);
    } catch (error: any) {
      const message = 
        error.response?.data?.error || 
        error.message || 
        'Registration failed - please try again';
      console.error('AuthService register error:', error);
      throw new Error(message);
    }
  }

  static async checkHealth(): Promise<boolean> {
    try {
      const response = await axios.get(`${BASE_URL}/health`);
      return response.status === 200;
    } catch {
      return false;
    }
  }
}
